clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -45,60 +45,78 @@
+        } else
+            struct.set(TOPICS_KEY_NAME, null);
+
         this.groupId = groupId;
         this.partitions = partitions;
     }
 
     public OffsetFetchRequest(Struct struct) {
         super(struct);
-        partitions = new ArrayList<>();
-        for (Object topicResponseObj : struct.getArray(TOPICS_KEY_NAME)) {
-            Struct topicResponse = (Struct) topicResponseObj;
-            String topic = topicResponse.getString(TOPIC_KEY_NAME);
-            for (Object partitionResponseObj : topicResponse.getArray(PARTITIONS_KEY_NAME)) {
-                Struct partitionResponse = (Struct) partitionResponseObj;
-                int partition = partitionResponse.getInt(PARTITION_KEY_NAME);
-                partitions.add(new TopicPartition(topic, partition));
+
+        Object[] topicArray = struct.getArray(TOPICS_KEY_NAME);
+        if (topicArray != null) {
+            partitions = new ArrayList<>();
+            for (Object topicResponseObj : struct.getArray(TOPICS_KEY_NAME)) {
+                Struct topicResponse = (Struct) topicResponseObj;
+                String topic = topicResponse.getString(TOPIC_KEY_NAME);
+                for (Object partitionResponseObj : topicResponse.getArray(PARTITIONS_KEY_NAME)) {
+                    Struct partitionResponse = (Struct) partitionResponseObj;
+                    int partition = partitionResponse.getInt(PARTITION_KEY_NAME);
+                    partitions.add(new TopicPartition(topic, partition));
+                }
             }
-        }
+        } else
+            partitions = null;
+
         groupId = struct.getString(GROUP_ID_KEY_NAME);
     }
 
     @Override
     public AbstractResponse getErrorResponse(int versionId, Throwable e) {
-        Map<TopicPartition, OffsetFetchResponse.PartitionData> responseData = new HashMap<>();
-
-        for (TopicPartition partition: partitions) {
-            responseData.put(partition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET,
-                    OffsetFetchResponse.NO_METADATA,
-                    Errors.forException(e).code()));
-        }
+        Map<TopicPartition, OffsetFetchResponse.PartitionData> responseData = new HashMap();
+        short errorCode = Errors.NONE.code();
+
+        if (partitions != null)
+            for (TopicPartition partition: partitions) {
+                responseData.put(partition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET,
+                        OffsetFetchResponse.NO_METADATA,
+                        Errors.forException(e).code()));
+            }
 
         switch (versionId) {
             // OffsetFetchResponseV0 == OffsetFetchResponseV1
             case 0:
             case 1:
-                return new OffsetFetchResponse(responseData);
+                return new OffsetFetchResponse(responseData, versionId);
+            case 2:
+                return new OffsetFetchResponse(responseData, errorCode);
```

Comment:
That's fair. I can modify the condition of the `if` block before this `switch` statement to skip building `responseData` for version 2 and beyond.